frontend/workflows/projectSelector/src/project-selector.tsx

```diff
@@ -63,28 +75,43 @@ const StyledProgressContainer = styled.div({
 });
 
 // Determines if every project has projectData (i.e. the effect has finished fetching the data)
-const allPresent = (state: State): boolean => {
-  let ret = true;
-  const allProjects = new Set([
-    ...Object.keys(state[Group.PROJECTS]),
-    ...Object.keys(state[Group.UPSTREAM]),
-    ...Object.keys(state[Group.DOWNSTREAM]),
-  ]);
-  allProjects.forEach(p => {
-    if (!(p in state.projectData) || _.isEmpty(state.projectData?.[p])) {
-      ret = false;
-    }
-    return ret; // Will stop iteration early if false encountered.
-  });
-  return ret;
+const allPresent = (state: State, dispatch: React.Dispatch<Action>): boolean => {
+  const projectCheck = project =>
+    project in state.projectData && !_.isEmpty(state.projectData?.[project]);
+
+  // check for data on all of our top level projects
+  const PROJECTS = Array.from(new Set(Object.keys(state[Group.PROJECTS])));
+  if (!PROJECTS.every(projectCheck)) {
+    // missing data for a project, return
+    return false;
+  }
+
+  const upDownKeys = Array.from(
+    new Set([...Object.keys(state[Group.UPSTREAM]), ...Object.keys(state[Group.UPSTREAM])])
+  );
+
+  // we've received all data for our projects, check the upstreams / downstreams
+  const missing = upDownKeys.filter(p => !projectCheck(p));
+  if (missing.length) {
+    // we're missing upstreams / downstreams, remove them from our state
+    dispatch({
+      type: "HYDRATE_DEPRECATION",
+      payload: { result: { missing, type: "STREAMS" } },
+    });
+  }
```

Comment:
for my understanding, why is there a `"HYDRATE_DEPRECATION` call here and after the API call

Reply:
Once we've loaded our projects there is a chance that a given upstream or downstream that was previously stored is also deprecated and will need to be removed from the state so that it can proceed

Reply:
Modified this logic to utilize HYDRATE_ERROR to remove missing projects from the state